src/frontend/screens/MapScreen/GPSPermissions/GPSPermissionsModal.tsx

```diff
@@ -1,34 +1,84 @@
-import React, {useEffect, useState} from 'react';
-import {GPSPermissionsDisabled} from './GPSPermissionsDisabled';
+import React from 'react';
+import {GPSForegroundPermissionDisabled} from './GPSForegroundPermissionDisabled';
 import {GPSPermissionsEnabled} from './GPSPermissionsEnabled';
 import * as Location from 'expo-location';
 import {useGPSModalContext} from '../../../contexts/GPSModalContext';
 import {useTabNavigationStore} from '../../../hooks/useTabNavigationStore';
 import {BottomSheetModal, BottomSheetView} from '@gorhom/bottom-sheet';
 import {TAB_BAR_HEIGHT} from '../../../Navigation/Stack/AppScreens';
-import {StyleSheet} from 'react-native';
-import {useFocusEffect} from '@react-navigation/native';
+import {StyleSheet, Linking, View} from 'react-native';
+import {GPSBackgroundPermissionDisabled} from './GPSBackgroundPermissionDisabled';
+import {Loading} from '../../../sharedComponents/Loading';
+
+const handleOpenSettings = () => {
+  Linking.sendIntent('android.settings.LOCATION_SOURCE_SETTINGS');
```

Comment:
if the user ends up going down this path, there's an issue where the app doesn't realize that the permission has been granted when returning to it. in order for the app to recognize that the permission is granted, they have to restart the app.

this is related to the issue you referred to in expo location, so i don't think there's a trivial way of properly handling this case. the "easy" short-term solution would be to inform the user that a restart would be needed.

don't think this is a blocker, but something to keep an eye on for later.